htap_test.go

```diff
@@ -409,3 +412,165 @@ func TestHybridTablesE2E(t *testing.T) {
 		})
 	})
 }
+
+func TestHTAPOptimizations(t *testing.T) {
+	if runningOnGithubAction() {
+		t.Skip("insufficient permissions")
+	}
+	for _, useHtapOptimizations := range []bool{true, false} {
+		runSnowflakeConnTest(t, func(sct *SCTest) {
+			t.Run("useHtapOptimizations="+strconv.FormatBool(useHtapOptimizations), func(t *testing.T) {
+				if useHtapOptimizations {
+					sct.mustExec("ALTER SESSION SET ENABLE_SNOW_654741_FOR_TESTING = true", nil)
```

Comment:
don't we want to defer setting this parameter back to false?

Reply:
No, we don't need, as this is only a session parameter. And especially - for local testing only.